### PR TITLE
Add converters

### PR DESCRIPTION
**Pull request by emillon**
This requires adding a few type aliases to make deriving work properly.

### CODE REVIEW

File: key_parsers.ml [hunk 1]
```diff
@@ -1,6 +1,23 @@
 let try_with_asn f = try Result.Ok (f ()) with Asn.Parse_error s -> Result.Error s
 let raise_asn f = match f () with Result.Ok x -> x | Result.Error s -> Asn.parse_error s
 
+let pp_of_to_string to_string fmt x =
+  Format.pp_print_string fmt (to_string x)
+
+type oid = Asn.OID.t
+let compare_oid a b =
+  String.compare (Asn.OID.to_string a) (Asn.OID.to_string b)
+let pp_oid = pp_of_to_string Asn.OID.to_string
```
Comment (NathanReb) [inline key_parsers.ml]: How about:

``` ocaml
module Z = struct
  include Z

  let pp = ...
end
```

Same applies to other occurences.

File: key_parsers.ml [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+
+type z = Z.t
+let compare_z = Z.compare
+let pp_z = pp_of_to_string Z.to_string
+
+type cstruct = Cstruct.t
+let compare_cstruct = Cstruct.compare
+let pp_cstruct =
+  pp_of_to_string (fun cs -> Printf.sprintf "%S" @@ Cstruct.to_string cs)
```
Comment (NathanReb) [inline key_parsers.ml]: We might want it to hex dump the string so it's human readable. What do you think ?

File: key_parsers.ml (hunk 1 above)
Comment (emillon) [inline key_parsers.ml]: Done! (this requires some tricks to be able to extend the submodule).

File: key_parsers.ml (hunk 2 above)
Comment (emillon) [inline key_parsers.ml]: Done, though `Cstruct.hexdump_to_buffer` contains some newlines, which may be what we want or not.

**Comment by emillon** [comment]
Just pushed a few exports, and then it's ready for me. Let me know what you think.

**Comment by NathanReb** [comment]
Looks good to me ! Merging

**Pull request merged**